sycl/include/CL/sycl/buffer.hpp

```diff
@@ -181,8 +181,10 @@ class buffer {
       : Range{0} {
 
     size_t BufSize = 0;
-    PI_CALL(piMemGetInfo)(detail::pi::cast<detail::RT::PiMem>(MemObject),
-                          CL_MEM_SIZE, sizeof(size_t), &BufSize, nullptr);
+    auto Plugin = detail::getSyclObjImpl(SyclContext)->getPlugin();
+    Plugin.call<detail::PiApiKind::piMemGetInfo>(
```

Comment:
@AlexeySachkov  any comments?

Reply:
> Also, what is the benefit over the existing call format?

In most cases in this PR, call to a plugin performed only once, so suggested format can save you one line at most of call sites.

> We wanted to avoid making such calls that takes an extra argument beyond the arguments of the function call.
> It should look like a function call.

I'm not pushing towards my suggestion, this was just an idea. If the intent was to make this call as close as possible to actual call which will be performed under the hood, then let it be. I don't have objections against it